### PR TITLE
Port log rework from next

### PR DESCRIPTION
**Pull request by krocard**
Port log rework (#87) from next in order to remove dependencies on *sprintf that does not have the same behavior on windows.

Additionally fixed a leak introduced by #87 (port make_unique for that)
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/168%23issuecomment-135088171%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23issuecomment-135354269%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38000600%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38001884%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38072772%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38072865%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38072999%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38073263%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38073710%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38074225%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38074987%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38075085%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38075259%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38127627%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38127815%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38128611%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38128707%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38128718%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38128726%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38128787%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38179964%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38180877%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38182642%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38182643%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38182644%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38182645%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/168%23issuecomment-135088171%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40dawagner%20%40OznOg%20please%20review.%22%2C%20%22created_at%22%3A%20%222015-08-26T16%3A23%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A-1%3A%20%22%2C%20%22created_at%22%3A%20%222015-08-27T09%3A11%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20939da499892979b8766cf2a9f25773fcdff556c6%20bindings/c/ParameterFramework.cpp%2028%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38000600%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22should%20not%20be%20in%20this%20pull%20request%22%2C%20%22created_at%22%3A%20%222015-08-26T16%3A06%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-08-26T16%3A17%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20bindings/c/ParameterFramework.cpp%3AL147-154%22%7D%2C%20%22Pull%200ad564401125e16d0aa4994ab302033a281b0ba9%20bindings/c/ParameterFramework.cpp%209%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38072772%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22should%20not%20be%20virtual%3F%22%2C%20%22created_at%22%3A%20%222015-08-27T08%3A29%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-08-27T18%3A23%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20bindings/c/ParameterFramework.cpp%3AL105-123%22%7D%2C%20%22Pull%200ad564401125e16d0aa4994ab302033a281b0ba9%20parameter/ParameterMgr.h%2093%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38074225%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22is%20that%20supposed%20to%20have%20a%20link%20with%20logs%3F%22%2C%20%22created_at%22%3A%20%222015-08-27T08%3A49%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22Yes%2C%20created%20a%20function%20to%20split%20the%20load%20function%20and%20avoid%20it%20to%20be%20too%20big.%20%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-08-28T08%3A46%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-08-28T09%3A10%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ParameterMgr.h%3AL530-543%22%7D%2C%20%22Pull%200ad564401125e16d0aa4994ab302033a281b0ba9%20test/test-platform/TestPlatform.cpp%2057%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38075259%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22wrapping%20seem%20useless%22%2C%20%22created_at%22%3A%20%222015-08-27T09%3A01%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22It%20was%20part%20of%20the%20original%20pull%20request%20and%20does%20not%20really%20matter%20%28wrapped%20at%2080%29.%20%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-08-27T18%3A14%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-08-28T09%3A10%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/test-platform/TestPlatform.cpp%3AL368-379%22%7D%2C%20%22Pull%200ad564401125e16d0aa4994ab302033a281b0ba9%20parameter/ConfigurableDomains.h%2027%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38182644%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-08-28T09%3A10%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ConfigurableDomains.h%3AL79-98%22%7D%2C%20%22Pull%200ad564401125e16d0aa4994ab302033a281b0ba9%20bindings/c/ParameterFramework.cpp%2029%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38072865%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22missing%20space%22%2C%20%22created_at%22%3A%20%222015-08-27T08%3A30%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-08-27T18%3A23%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20bindings/c/ParameterFramework.cpp%3AL199-209%22%7D%2C%20%22Pull%200ad564401125e16d0aa4994ab302033a281b0ba9%20parameter/log/include/log/LogWrapper.h%2075%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38074987%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22missing%20space%22%2C%20%22created_at%22%3A%20%222015-08-27T08%3A58%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22Done%20%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-08-28T08%3A32%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/log/include/log/LogWrapper.h%3AL1-138%22%7D%2C%20%22Pull%200ad564401125e16d0aa4994ab302033a281b0ba9%20parameter/log/include/log/LogWrapper.h%20108%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38075085%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22coding%20style%22%2C%20%22created_at%22%3A%20%222015-08-27T08%3A59%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20guess%20you%20were%20talking%20about%20the%20-1%20%3F%20Fixed%20%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-08-27T18%3A16%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/log/include/log/LogWrapper.h%3AL1-138%22%7D%2C%20%22Pull%200ad564401125e16d0aa4994ab302033a281b0ba9%20parameter/ParameterMgr.cpp%20208%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38073710%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22missing%20space%22%2C%20%22created_at%22%3A%20%222015-08-27T08%3A42%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-08-27T18%3A24%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ParameterMgr.cpp%3AL524-564%22%7D%2C%20%22Pull%200ad564401125e16d0aa4994ab302033a281b0ba9%20parameter/ConfigurableDomains.h%2052%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38182645%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-08-28T09%3A10%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ConfigurableDomains.h%3AL100-122%22%7D%2C%20%22Pull%200ad564401125e16d0aa4994ab302033a281b0ba9%20parameter/ConfigurableDomains.cpp%20167%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38073263%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22keep%20bool%20on%20the%20same%20line%22%2C%20%22created_at%22%3A%20%222015-08-27T08%3A36%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-08-27T18%3A23%3A45Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ConfigurableDomains.cpp%3AL517-540%22%7D%2C%20%22Pull%200ad564401125e16d0aa4994ab302033a281b0ba9%20doc/CMakeLists.txt%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/168%23discussion_r38072999%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22seems%20strange%20in%20here%22%2C%20%22created_at%22%3A%20%222015-08-27T08%3A32%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22Restored%20%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-08-27T18%3A22%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20doc/CMakeLists.txt%3AL58-61%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/168#issuecomment-135088171'>General Comment</a></b>
- <a href='https://github.com/krocard'></a> @dawagner @OznOg please review.
- <a href='https://github.com/OznOg'></a> :-1:
- [x] <a href='#crh-comment-Pull 939da499892979b8766cf2a9f25773fcdff556c6 bindings/c/ParameterFramework.cpp 28'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/168#discussion_r38000600'>File: bindings/c/ParameterFramework.cpp:L147-154</a></b>
- <a href='https://github.com/krocard'></a> should not be in this pull request
- [x] <a href='#crh-comment-Pull 0ad564401125e16d0aa4994ab302033a281b0ba9 bindings/c/ParameterFramework.cpp 9'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/168#discussion_r38072772'>File: bindings/c/ParameterFramework.cpp:L105-123</a></b>
- <a href='https://github.com/OznOg'></a> should not be virtual?
- [x] <a href='#crh-comment-Pull 0ad564401125e16d0aa4994ab302033a281b0ba9 bindings/c/ParameterFramework.cpp 29'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/168#discussion_r38072865'>File: bindings/c/ParameterFramework.cpp:L199-209</a></b>
- <a href='https://github.com/OznOg'></a> missing space
- [x] <a href='#crh-comment-Pull 0ad564401125e16d0aa4994ab302033a281b0ba9 doc/CMakeLists.txt 5'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/168#discussion_r38072999'>File: doc/CMakeLists.txt:L58-61</a></b>
- <a href='https://github.com/OznOg'></a> seems strange in here
- <a href='https://github.com/krocard'></a> Restored :+1:
- [x] <a href='#crh-comment-Pull 0ad564401125e16d0aa4994ab302033a281b0ba9 parameter/ConfigurableDomains.cpp 167'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/168#discussion_r38073263'>File: parameter/ConfigurableDomains.cpp:L517-540</a></b>
- <a href='https://github.com/OznOg'></a> keep bool on the same line
- [x] <a href='#crh-comment-Pull 0ad564401125e16d0aa4994ab302033a281b0ba9 parameter/ParameterMgr.cpp 208'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/168#discussion_r38073710'>File: parameter/ParameterMgr.cpp:L524-564</a></b>
- <a href='https://github.com/OznOg'></a> missing space
- [x] <a href='#crh-comment-Pull 0ad564401125e16d0aa4994ab302033a281b0ba9 parameter/ParameterMgr.h 93'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/168#discussion_r38074225'>File: parameter/ParameterMgr.h:L530-543</a></b>
- <a href='https://github.com/OznOg'></a> is that supposed to have a link with logs?
- <a href='https://github.com/krocard'></a> Yes, created a function to split the load function and avoid it to be too big. :+1:
- [x] <a href='#crh-comment-Pull 0ad564401125e16d0aa4994ab302033a281b0ba9 parameter/log/include/log/LogWrapper.h 75'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/168#discussion_r38074987'>File: parameter/log/include/log/LogWrapper.h:L1-138</a></b>
- <a href='https://github.com/OznOg'></a> missing space
- <a href='https://github.com/krocard'></a> Done :+1:
- [x] <a href='#crh-comment-Pull 0ad564401125e16d0aa4994ab302033a281b0ba9 parameter/log/include/log/LogWrapper.h 108'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/168#discussion_r38075085'>File: parameter/log/include/log/LogWrapper.h:L1-138</a></b>
- <a href='https://github.com/OznOg'></a> coding style
- <a href='https://github.com/krocard'></a> I guess you were talking about the -1 ? Fixed :+1:
- [x] <a href='#crh-comment-Pull 0ad564401125e16d0aa4994ab302033a281b0ba9 test/test-platform/TestPlatform.cpp 57'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/168#discussion_r38075259'>File: test/test-platform/TestPlatform.cpp:L368-379</a></b>
- <a href='https://github.com/OznOg'></a> wrapping seem useless
- <a href='https://github.com/krocard'></a> It was part of the original pull request and does not really matter (wrapped at 80). :+1:

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/168?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/168?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/168'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/168'>Refresh</a>.<a href='#crh-update-none'></a>

File: bindings/c/ParameterFramework.cpp [hunk 1]
```diff
@@ -144,7 +147,7 @@ PfwHandler *pfwCreate()
 
 void pfwDestroy(PfwHandler *handle)
 {
-    if(handle != NULL and handle->pfw != NULL) {
+    if(handle != NULL && handle->pfw != NULL) {
```
Comment (krocard) [inline bindings/c/ParameterFramework.cpp]: should not be in this pull request

Comment (krocard) [inline bindings/c/ParameterFramework.cpp]: :+1:

**Comment by krocard** [comment]
@dawagner @OznOg please review.

File: bindings/c/ParameterFramework.cpp [hunk 2]
```diff
@@ -105,15 +105,18 @@ class LogWrapper : public CParameterMgrPlatformConnector::ILogger
     LogWrapper() : mLogger() {}
     virtual ~LogWrapper() {}
 private:
-    virtual void log(bool bIsWarning, const string &strLog)
+    virtual void info(const string &msg) { log(pfwLogInfo, msg); }
+
+    virtual void warning(const string &msg) { log(pfwLogWarning, msg); }
+
+    virtual void log(PfwLogLevel level, const string &strLog)
```
Comment (OznOg) [inline bindings/c/ParameterFramework.cpp]: should not be virtual?

File: bindings/c/ParameterFramework.cpp [hunk 3]
```diff
@@ -196,9 +199,10 @@ bool PfwHandler::createCriteria(const PfwCriterion criteriaArray[], size_t crite
                 value = valueIndex;
             }
             const char * valueName = criterion.values[valueIndex];
-            if(not type->addValuePair(value, valueName)) {
+            string error;
+            if(not type->addValuePair(value, valueName, error)) {
```
Comment (OznOg) [inline bindings/c/ParameterFramework.cpp]: missing space

File: doc/CMakeLists.txt [hunk 4]
```diff
@@ -58,5 +58,3 @@ if(DOXYGEN)
         COMMENT "Generating documentation with Doxygen"
         VERBATIM)
 endif()
-
-add_subdirectory(requirements)
```
Comment (OznOg) [inline doc/CMakeLists.txt]: seems strange in here

File: parameter/ConfigurableDomains.cpp [hunk 5]
```diff
@@ -510,17 +517,23 @@ void CConfigurableDomains::listLastAppliedConfigurations(string& strResult) cons
 }
 
 // Configurable element - domain association
-bool CConfigurableDomains::addConfigurableElementToDomain(const string& strDomain, CConfigurableElement* pConfigurableElement, const CParameterBlackboard* pMainBlackboard, string& strError)
+bool
+CConfigurableDomains::addConfigurableElementToDomain(const string& domainName,
+                                                     CConfigurableElement* element,
```
Comment (OznOg) [inline parameter/ConfigurableDomains.cpp]: keep bool on the same line

File: parameter/ConfigurableDomains.h [hunk 6]
```diff
@@ -79,7 +79,19 @@ class CConfigurableDomains : public CBinarySerializableElement
     bool setSequenceAwareness(const std::string& strDomain, bool bSequenceAware, std::string& strError);
     bool getSequenceAwareness(const std::string& strDomain, bool& bSequenceAware, std::string& strError) const;
     bool listDomainElements(const std::string& strDomain, std::string& strResult) const;
-    bool split(const std::string& strDomain, CConfigurableElement* pConfigurableElement, std::string& strError);
+
+    /** Split a domain in two.
+     * Remove an element of a domain and create a new domain which owns the element.
+     *
+     * @param[in] domainName the domain name
+     * @param[in] element pointer to the element to remove
```
Comment (OznOg) [inline parameter/ConfigurableDomains.h]: seems to have nothing to do with logs

File: parameter/ConfigurableDomains.h [hunk 7]
```diff
@@ -88,7 +100,22 @@ class CConfigurableDomains : public CBinarySerializableElement
...
+
+    /** Restore a configuration
+     *
+     * @param[in] strDomain the domain name
+     * @param[in] strConfiguration the configuration name
+     * @param[in] mainBlackboard the application main blackboard
+     * @param[in] autoSync boolean which indicates if auto sync mechanism is on
+     * @param[out] errors, errors encountered during restoration
+     * @return true if success false otherwise
+     */
```
Comment (OznOg) [inline parameter/ConfigurableDomains.h]: idem

File: parameter/ParameterMgr.cpp [hunk 8]
```diff
@@ -554,17 +524,40 @@ bool CParameterMgr::loadFrameworkConfiguration(string& strError)
...
+
+    // Load subsystems
+    bool isSuccess = getSystemClass()->loadSubsystems(error,
+                                                      _pSubsystemPlugins,
+                                                      !_bFailOnMissingSubsystem);
+
+    if (isSuccess) {
+        info() << "All subsystem plugins successfully loaded";
+
+        if(!error.empty()) {
```
Comment (OznOg) [inline parameter/ParameterMgr.cpp]: missing space

File: parameter/ParameterMgr.h [hunk 9]
```diff
@@ -534,6 +530,13 @@ class CParameterMgr : private CElement
     // Framework global configuration loading
     bool loadFrameworkConfiguration(std::string& strError);
 
+    /** Load required subsystems
+     *
```
Comment (OznOg) [inline parameter/ParameterMgr.h]: is that supposed to have a link with logs?

File: parameter/log/include/log/LogWrapper.h [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+     */
+    LogWrapper(const LogWrapper& logWrapper)
+        : mLogger(logWrapper.mLogger), mProlog(logWrapper.mProlog)
+    {}
+
+    /** Class destructor */
+    ~LogWrapper()
+    {
+        if (!mLog.str().empty()) {
+            if(isWarning) {
```
Comment (OznOg) [inline parameter/log/include/log/LogWrapper.h]: missing space

File: parameter/log/include/log/LogWrapper.h [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+     * @param[in] logs list of information to log
+     */
+    LogWrapper& operator<<(const std::list<std::string>& logs)
+    {
+        std::string formatedLogs;
+        std::string separator = "\n" + mProlog;
+        CUtility::asString(logs, formatedLogs, separator);
+
+        // Check if there is something in the log to know if we have to add a prefix
+        if (!mLog.str().empty() && mLog.str()[mLog.str().length()-1] == separator[0]) {
```
Comment (OznOg) [inline parameter/log/include/log/LogWrapper.h]: coding style

File: test/test-platform/TestPlatform.cpp [hunk 12]
```diff
@@ -369,9 +368,11 @@ bool CTestPlatform::createExclusiveSelectionCriterion(const string& strName,
         ostrValue << "State_";
         ostrValue << uistate;
 
-        if (!pCriterionType->addValuePair(uistate, ostrValue.str())) {
+        if (!pCriterionType->addValuePair(
+                    uistate, ostrValue.str(), strResult)) {
 
-            strResult = "Unable to add value: " + ostrValue.str();
+            strResult = "Unable to add value: "
```
Comment (OznOg) [inline test/test-platform/TestPlatform.cpp]: wrapping seem useless

**Comment by OznOg** [comment]
:-1:

File: test/test-platform/TestPlatform.cpp (hunk 12 above)
Comment (krocard) [inline test/test-platform/TestPlatform.cpp]: It was part of the original pull request and does not really matter (wrapped at 80).

File: parameter/log/include/log/LogWrapper.h (hunk 11 above)
Comment (krocard) [inline parameter/log/include/log/LogWrapper.h]: I guess you were talking about the -1 ? Fixed :+1:

File: doc/CMakeLists.txt (hunk 4 above)
Comment (krocard) [inline doc/CMakeLists.txt]: Restored :+1:

File: bindings/c/ParameterFramework.cpp (hunk 3 above)
Comment (krocard) [inline bindings/c/ParameterFramework.cpp]: :+1:

File: bindings/c/ParameterFramework.cpp (hunk 2 above)
Comment (krocard) [inline bindings/c/ParameterFramework.cpp]: :+1:

File: parameter/ConfigurableDomains.cpp (hunk 5 above)
Comment (krocard) [inline parameter/ConfigurableDomains.cpp]: :+1:

File: parameter/ParameterMgr.cpp (hunk 8 above)
Comment (krocard) [inline parameter/ParameterMgr.cpp]: :+1:

File: parameter/log/include/log/LogWrapper.h (hunk 10 above)
Comment (krocard) [inline parameter/log/include/log/LogWrapper.h]: Done :+1:

File: parameter/ParameterMgr.h (hunk 9 above)
Comment (krocard) [inline parameter/ParameterMgr.h]: Yes, created a function to split the load function and avoid it to be too big.

Comment (krocard) [inline parameter/ParameterMgr.h]: :+1:<a href='#crh-update-none'></a>

File: test/test-platform/TestPlatform.cpp (hunk 12 above)
Comment (krocard) [inline test/test-platform/TestPlatform.cpp]: :+1:<a href='#crh-update-none'></a>

File: parameter/ConfigurableDomains.h (hunk 6 above)
Comment (krocard) [inline parameter/ConfigurableDomains.h]: :+1:<a href='#crh-update-none'></a>

File: parameter/ConfigurableDomains.h (hunk 7 above)
Comment (krocard) [inline parameter/ConfigurableDomains.h]: :+1:<a href='#crh-update-complete'></a>

**Pull request merged**